### PR TITLE
feat(component): icon button

### LINKED ISSUE #930: IconButton
### Short Description
A standard `Button` that only displays an icon (with a required, accessible label).

### Mockup

[Example](https://github.com/narmi/banking/pull/31592#discussion_r1188976403)

### Responsibilities
Defaults for all props.

**This component should be responsible for the following:**
- Render an icon by name
- Adding an `aria-label`
- Using same size and color variant options as `Button`
- Default to `kind="plain"` (without underline)
- Padding options / click target sizing (unclear if it 

**This component should NOT be responsible for...**
- Rendering any kind of visible label
- `as` prop (it should always render a button)
- 

### Props Signature

```
<IconButton
  size={standard sizes}
  kind={standard Button kinds}
  paddingSize?={standard size tokens}
  name={nds icon name}
  label="REQUIRED LABEL"
/>
```

### Open questions

Some of these will require a discussion with design:

- Should we support multiple variants (`kind`) of `IconButton`, or just "plain"?
- Are hover states consistent across icon buttons in design?
- Should the padding be configurable, or controlled by `size` prop?
- Do we need to have an option to expand the click target area out of document flow?

### PR DESCRIPTION
**Pull request by JohnathanWeisner**
resolves #930

Adds an IconButton component.

### CODE REVIEW

File: src/IconButton/index.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  testId = "nds-icon-button",
+  label,
+  name,
+  onClick = () => {},
+  className,
+  ...otherProps
+}) => {
+  return (
+    <button
+      tabIndex={0}
```
Comment (akdetrick) [inline src/IconButton/index.js:31]: we get this for free with a `button` element
```suggestion
```

File: src/IconButton/index.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      onClick={onClick}
+      {...otherProps}
```
Comment (akdetrick) [inline src/IconButton/index.js:33]: We generally avoid spread props on NDS components except for form elements. 

The idea is that we want to prevent `style` and `className` from being passed - if there's a styling change we need to make in product, we should work with design to come up with a standard variant instead of accepting arbitrary styles.

In this case, your `className` prop comes after spread props, so we may be fine if `style` is plucked from `...otherProps`

File: src/IconButton/index.scss [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  outline: inherit;
+  font-size: var(--font-size-default);
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  border-radius: var(--button-radius);
+  height: 32px;
+  width: 32px;
+  /* mobile buttons are larger for ease of tapping */
+  @media (max-width: #{map.get($breakpoints, "s")}) {
```
Comment (akdetrick) [inline src/IconButton/index.scss:17]: Optional suggestion - we have a Sass helper to do this in a mobile-first way:

```
    /* small is "default" */
    height: 48px;
    width: 48px;;
    
    /* once the viewport reaches medium OR greater, apply these styles: */
    @include atMediaUp("m") {
        height: 32px;
        width: 32px;
    }
```

File: src/IconButton/index.scss [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    height: 48px;
+    width: 48px;
+  }
+
+  /* Nudge the large size icons to be centered more accurately */
+  &.fontSize--l {
+    [class^="narmi-icon-"] {
+      position: relative;
```
Comment (akdetrick) [inline src/IconButton/index.scss:25]: would recommend translating instead, but not a huge deal to keep it this way

`transform: translate(-1px, -0.5px);`

Translate accomplishes the same thing without any performance hits if for some reason we want to animate hover or active states

**Review by akdetrick — COMMENTED** [review]
This is really nice, thanks!

Before I approve, I'm going to check in with design on:
- variants needed
- design details (I agree with your choices)
- if we need to support a visual "active" state

**Review by akdetrick — CHANGES_REQUESTED** [review]
Updates from design:

https://github.com/narmi/design_system/issues/930#issuecomment-1608110313

**tl;dr**
- `kind: PropTypes.oneOf(["action", "plain"]),`
- direct size control should be removed for the initial version of this component
- mocks/specs are in the comment linked above

File: src/IconButton/index.scss [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  &.fontSize--l {
+    [class^="narmi-icon-"] {
+      transform: translate(0px, -1px);
+    }
+  }
+}
+
+.nds-icon-button--action {
+  --button-radius: 50%;
+  color: var(--color-secondary);
```
Comment (akdetrick) [inline src/IconButton/index.scss:24]: Initial color should be secondary text color:

```suggestion
  color: var(--font-color-secondary);
```

File: src/IconButton/index.scss [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    color: var(--theme-primary);
+    background-color: rgba(var(--theme-rgb-primary), var(--alpha-5));
+    text-decoration: none;
+  }
+}
+
+.nds-icon-button--plain {
+  height: 24px;
+  width: 24px;
+  color: var(--theme-secondary);
```
Comment (akdetrick) [inline src/IconButton/index.scss:36]: Same initial color for plain buttons:

```suggestion
  color: var(--theme-secondary);
```

File: src/IconButton/index.js [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+  /** Name of Narmi icon */
+  name: PropTypes.oneOf(VALID_ICON_NAMES).isRequired,
+  /** Renders the icon button label */
+  label: PropTypes.string,
+  /** disables the icon button when set to `true` */
+  disabled: PropTypes.bool,
+  /** style of icon button to render */
+  kind: PropTypes.oneOf(["action", "plain"]),
+  /** Optional text size of the icon in the icon button defaults different for different kinds (plain/action)*/
+  textSize: PropTypes.oneOf(["xs", "s", "m", "l"]),
```
Comment (akdetrick) [inline src/IconButton/index.js:58]: What do you think about eliminating `xs` and maybe even `s` as valid options?

I don't think we ever want to go below "m" text, right?

**Review by akdetrick — CHANGES_REQUESTED** [review]
Looks good! Just a few minor things before we merge

File: src/IconButton/index.js (hunk 7 above)
Comment (JohnathanWeisner) [reply src/IconButton/index.js:58]: Unfortunately the info icon is `xs` size in the figma.

**Review by akdetrick — APPROVED** [review]
🚀
